scripts/apps/search/components/Item.tsx

```diff
@@ -119,13 +120,21 @@ export class Item extends React.Component<IProps, IState> {
         }
     }
 
+    handleHighlightsLoading(e: CustomEvent) {
+        if (e.detail.itemId === this.props.item._id) {
+            this.setState({loading: e.detail.loading});
+        }
+    }
+
     componentDidMount() {
+        addEventListener('highlights-loading', this.handleHighlightsLoading);
```

Comment:
Let's not couple it to highlights here. Just call it loading indicator. That way we'll be able to reuse it for other actions and will not be lying via naming.